app/Models/Feature/Feature.php

```diff
@@ -104,6 +104,14 @@ public function parent()
         return $this->belongsTo('App\Models\Feature\Feature', 'parent_id');
     }
 
+    /**
+     * Get the parent of this feature, if present and visible.
+     */
+    public function parentVisible()
+    {
+        return $this->belongsTo('App\Models\Feature\Feature', 'parent_id')->where('is_visible', 1);
```

Comment:
This looks to only be used once in a way that'd be more easily done otherwise (see related comment below), so I would just omit it entirely.

Reply:
Removed, as no longer necessary with changes!